internal/controllers/clusterrequest_controller.go

```diff
@@ -1135,6 +1140,9 @@ func (t *clusterRequestReconcilerTask) updateClusterProvisionStatus(ci *siteconf
 				metav1.ConditionUnknown,
 				"Waiting for cluster provisioning to start",
 			)
+			if t.object.Status.ClusterDetails.ClusterProvisionStartedAt.IsZero() {
+				t.object.Status.ClusterDetails.ClusterProvisionStartedAt = metav1.Now()
+			}
```

Comment:
Since this code is common to both cases, consider moving it outside the if-else clause.

Reply:
There is another nested if condition, where it's only be set when clusterInstance provision is not present but clusterInstanceProcessed is completed.